plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/schema/SQLSchemaManager.java

```diff
@@ -87,6 +88,7 @@ public void updateSchema(DBRProgressMonitor monitor) throws DBException {
             try (JDBCTransaction txn = new JDBCTransaction(dbCon)) {
                 try {
                     int currentSchemaVersion = versionManager.getCurrentSchemaVersion(monitor, dbCon, targetSchemaName);
+                    BackupRegistry.getInstance().getSettings().forEach(p -> p.getInstance().doBackup(dbCon, currentSchemaVersion));
```

Comment:
do backup only if schema was changed

Reply:
why forEach?

Reply:
only one should be used if it is suitable for the current driver/dialect/other reasons